### PR TITLE
Make facet paginator class configurable.

### PR DESCRIPTION
**Pull request by rwd**
Facet pagination is still coupled to Solr, which will be problematic for installations of Blacklight using a different data source.

This commit takes the first step towards decoupling by making the facet paginator class configurable, and moves facet pagination query assembly into Solr's search builder.

### CODE REVIEW

**Comment by cbeer** [comment]
:+1:

**Comment by cbeer** [comment]
@rwd oh, and definitely feel free to ignore hound on code you haven't authored or modified in any substantial way.

**Comment by rwd** [comment]
@cbeer, yep, I figured that was fair enough where I've just copied and pasted while refactoring.

File: lib/blacklight/solr/search_builder_behavior.rb [hunk 1]
```diff
@@ -190,6 +194,30 @@ def add_group_config_to_solr solr_parameters
...
+        elsif solr_params["facet.limit"]
+          solr_params["facet.limit"].to_i
+        else
+          20
+        end
+
+      # Need to set as f.facet_field.facet.* to make sure we
+      # override any field-specific default in the solr request handler.
+      solr_params[:"f.#{facet}.facet.limit"]  = limit + 1
+      solr_params[:"f.#{facet}.facet.offset"] = ( blacklight_params.fetch(blacklight_config.facet_paginator_class.request_keys[:page] , 1).to_i - 1 ) * ( limit )
```
Comment (jcoyne) [inline lib/blacklight/solr/search_builder_behavior.rb]: Fix this one please. Then good too merge.

**Comment by rwd** [comment]
@cbeer, @jcoyne: are you happy to merge this now?

**Pull request merged**